### DISCUSSION TITLE
VisibilityChange Delay

### DISCUSSION THREAD

**Discussion by zkisaboss** [violentmonkey/violentmonkey#2210] [Q&A] [answered]
There's a delay in the visibility event listener. Not sure if it's caused by Firefox or Violentmonkey.

```
// ==UserScript==
// @name         Auto Play/Pause
// @version      1.0
// @icon         https://hianime.to/images/icons-192.png
// @author       Zach
// @match        https://megacloud.tv/*
// @grant        none
// ==/UserScript==

document.addEventListener("visibilitychange", () => {
    if (document.visibilityState === 'hidden') {
        jwplayer().pause();
    } else {
        jwplayer().play();
    }
```

**Comment by tophf** [comment, marked as answer]
Violentmonkey doesn't change behavior of standard events. You can debug what happens by eliminating the external code (jwplayer), which may have its own complicated logic. Replace it with console.log and see what happens. Compare behavior to other extensions and to other browsers e.g. Chromium.

**Reply by zkisaboss** [reply]
I just tried the same script on multiple browsers and Firefox is the only one affected. I'll try my luck reporting the issue to firefox.

**Reply by tophf** [reply]
When reporting, try to minimize the influence of extensions like Violentmonkey or things like jwplayer and run your code in devtools console with simple console.log() calls.
